### PR TITLE
Implemented some neat graph components and ability to export json and graph image

### PR DESCRIPTION
**Pull request by ds161833**
image

image

### CODE REVIEW

File: Client/src/Components/Graph/DatasetsList.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { Box, Grid } from "@material-ui/core"
+
+import { DatasetRow } from "./DatasetRow"
+import { IDatasetModel } from "../../Models/Datasets/IDatasetModel"
+import React from 'react'
+
+interface IProps {
+  datasets: IDatasetModel[],
+  onRemoveDatasetClick: (datasetId: number) => void
```
Comment (GD-Dheer) [inline Client/src/Components/Graph/DatasetsList.tsx:9]: Can we move the interfaces elsewhere and import? Reduces duplicate code

File: Client/src/Components/Search/SearchDatasetsForm.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import { Button, FormControl, FormControlLabel, Grid, InputLabel, Select, Switch, TextField, Typography } from '@material-ui/core'
+import { ISearchDatasetsFormModel, defaultSearchDatasetsModel } from '../../Models/Forms/ISearchDatasetsFormModel'
+import React, { useEffect, useState } from 'react'
+
+interface IProps {
+  handleSubmit(formValues: ISearchDatasetsFormModel): void
+}
+
+export const SearchDatasetsForm = (props: IProps): any => {
+  const [formValues, setFormValues] = useState(defaultSearchDatasetsModel)
```
Comment (GD-Dheer) [inline Client/src/Components/Search/SearchDatasetsForm.tsx:10]: Nice. Didnt even know you could use the hook like that

File: Client/src/Components/Search/SearchResults.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  //             <TableCell>Fuel</TableCell>
+  //             <TableCell>Diluent</TableCell>
+  //             <TableCell>Author</TableCell>
+  //             <TableCell>Year</TableCell>
+  //             <TableCell>Output Format</TableCell>
+  //           </TableRow>
+  //         </TableHead>
+  //         <TableBody>
+  //           {renderRows(props && props.datasetResults)}
+  //         </TableBody>
```
Comment (GD-Dheer) [inline Client/src/Components/Search/SearchResults.tsx]: Do we still need to keep this commented code?

**Review by GD-Dheer — COMMENTED** [review]
Great Work! Saw what it looked like earlier. A+

File: Client/src/Components/Search/SearchResults.tsx (hunk 3 above)
Comment (ds161833) [reply Client/src/Components/Search/SearchResults.tsx]: no, will remove 👍

File: Client/src/Components/Graph/DatasetsList.tsx (hunk 1 above)
Comment (ds161833) [reply Client/src/Components/Graph/DatasetsList.tsx:9]: thing is this interface is internal to this component and is not meant to be reusable, we can have a clean convention that every component that has props, has an IProps interface inside of it at the top of the file. This interface is created specifically for the needs of this component. The interfaces that represent some data model or abstract idea have their own files and are exported as you can see in my IDatasetModel.ts file.

**Review by GD-Dheer — APPROVED** [review]
Just need to resolve merge conflict

**Review by TiraFire — CHANGES_REQUESTED** [review]
As mentioned, the file export to JSON must follow the json format from the file upload (found in the story #77 and #127), this is the only change before approval, otherwise you did A LOT and the UI looks amazing. 

Make sure to separate your actions (ESLint should not be put inside this story), the refactor to TS for many of the files is a bonus (but again, make sure it is atleast documented inside the PR)
Obviously wont have you retract any of it but just for next time

File: Client/src/Components/Search/SearchResults.tsx [hunk 4]
```diff
@@ -39,30 +39,4 @@ export const SearchResults = (props: IProps) => {
       </Grid> : null}
     </Grid>
   )
-
```
Comment (TiraFire) [inline Client/src/Components/Search/SearchResults.tsx]: why is all this commented out?

File: Client/src/Models/Datasets/IExportDatasetModel.ts [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+  reference: IReference;
+  dataset_name: string;
+  material: IMaterial[];
+  category: string;
+  subcategory: string;
+  data_type: string;
+  data: IData;
+}
+
+export const exampleExportDatasetModel: IExportDatasetModel = {
```
Comment (TiraFire) [inline Client/src/Models/Datasets/IExportDatasetModel.ts:49]: Thank you for the proper file example ! thumbs up

File: Client/src/Views/GraphView.tsx [hunk 6]
```diff
@@ -8,8 +8,13 @@ import React, { useState } from "react"
 
 import { DatasetsList } from "../Components/Graph/DatasetsList"
 import Graph from '../Components/Graph/Graph'
-import { IDatasetModel } from "../Models/Datasets/IDatasetModel"
+import { ICompleteDatasetEntity } from "../Models/Datasets/ICompleteDatasetEntity"
+import { IGraphDatasetModel } from '../Models/Datasets/IGraphDatasetModel'
 import SearchView from './SearchView'
+import { exampleExportDatasetModel } from '../Models/Datasets/IExportDatasetModel'
+
+//todo this is poorly hardcoded, we need to let user set their own colors, as well as support more than just 4 colors.
```
Comment (TiraFire) [inline Client/src/Views/GraphView.tsx:16]: The array itself is just to keep track of the colours according to the array of datasets, once a better mapping is put into place, the set values will be changeable, as well as having more than just 4 datasets per graph, thank you for this observation and comment

**Review by TiraFire — APPROVED** [review]
Thanks for the changes, it looks good.

**Pull request merged**